code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -464,3 +434,335 @@
+
+	invisible_man.update_body()
+	invisible_man.update_hair()
+	invisible_man.remove_from_all_data_huds()
+	invisible_man.sound_environment_override = SOUND_ENVIROMENT_PHASED
+
+/datum/reagent/drug/saturnx/on_mob_end_metabolize(mob/living/M)
+	. = ..()
+	if(HAS_TRAIT(M, TRAIT_INVISIBLE_MAN))
+		M.add_to_all_human_data_huds() //Is this safe, what do you think, Floyd?
+		REMOVE_TRAIT(M, TRAIT_INVISIBLE_MAN, name)
+		to_chat(M, span_notice("As you sober up, opacity once again returns to your body meats."))
+
+		var/datum/dna/druggy_dna = M.has_dna()
+		if(druggy_dna?.species)
+			druggy_dna.species.species_traits -= NOBLOODOVERLAY
+
+	M.update_body()
+	M.update_hair()
+	M.sound_environment_override = NONE
+
+	if(!M.hud_used)
+		return
+
+	var/atom/movable/plane_master_controller/game_plane_master_controller = M.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+	game_plane_master_controller.remove_filter("saturnx_filter")
+	game_plane_master_controller.remove_filter("saturnx_blur")
+
+/datum/reagent/drug/saturnx/overdose_process(mob/living/M, delta_time, times_fired)
+	. = ..()
+	if(DT_PROB(7.5, delta_time))
+		M.emote("giggle")
+	if(DT_PROB(5, delta_time))
+		M.emote("laugh")
+	M.adjustOrganLoss(ORGAN_SLOT_LIVER, 0.4 * REM * delta_time)
+
+/datum/reagent/drug/kronkaine
+	name = "kronkaine"
+	description = "A highly illegal stimulant from the edge of the galaxy.\nIt is said the average kronkaine addict causes as much criminal damage as five stick up men, two rascals and one proferssional cambringo hustler combined."
+	reagent_state = SOLID
+	color = "#FAFAFA"
+	taste_description = "numbing bitterness"
+	ph = 8
+	overdose_threshold = 20
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+	addiction_types = list(/datum/addiction/stimulants = 20)
+
+/datum/reagent/drug/kronkaine/on_mob_metabolize(mob/living/L)
+	..()
+	L.add_actionspeed_modifier(/datum/actionspeed_modifier/kronkaine)
+	L.sound_environment_override = SOUND_ENVIRONMENT_HANGAR
+
+/datum/reagent/drug/kronkaine/on_mob_end_metabolize(mob/living/L)
+	L.remove_actionspeed_modifier(/datum/actionspeed_modifier/kronkaine)
+	L.sound_environment_override = NONE
+	. = ..()
+
+/datum/reagent/drug/kronkaine/on_transfer(atom/A, methods, trans_volume)
+	. = ..()
+	if(!iscarbon(A))
+		return
+	var/mob/living/carbon/druggo = A
+	druggo.adjustStaminaLoss(-4 * trans_volume, 0)
+	//I wish i could give it some kind of bonus when smoked, but we don't have an INHALE method.
+
+/datum/reagent/drug/kronkaine/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
+	. = ..()
+	SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "tweaking", /datum/mood_event/stimulant_medium, name)
+	M.adjustOrganLoss(ORGAN_SLOT_HEART, 0.4 * REM * delta_time)
+	M.Jitter(10 * REM * delta_time)
+	M.AdjustSleeping(-20 * REM * delta_time)
+	M.drowsyness = max(M.drowsyness - (5 * REM * delta_time), 0)
+	if(volume < 10)
+		return
+	for(var/possible_purger as anything in M.reagents.reagent_list)
```

Comment:
misuse of `as anything`. `as anything` is used to bypass byond's list filtering system (looking for whatever you typecast the var as) but you have not typecasted `possible_purger` as anything